dlt/pipeline/pipeline.py

```diff
@@ -878,6 +879,7 @@ def _extract_source(self, storage: ExtractorStorage, source: DltSource, max_para
         # note we are not merging props like max nesting or column propagation
         for table in source_schema.data_tables(include_incomplete=True):
             pipeline_schema.update_schema(pipeline_schema.normalize_table_identifiers(table))
```

Comment:
ok this is very interesting place:
we actually create and change schemas in the extract stage.
currently we allow for that to happen freely. we do not apply data contract. to be precise. stuff changes when
- someone adds new resource
- someone is using apply_hints
- we create a new table based on data

should we apply data contract for the last case?

Reply:
I have tests for adding new tables from a new resource and adding new tables as a subtable, and they both pass:
def test_freeze_new_tables(contract_setting: str, setting_location: str) -> None:
so it seems this case is covered already?

Reply:
I think you are not testing a case when 
1. table name is a lambda (dynamic) - see getting started with github_events. in that case you may get new tables when there's new event type. and I'm sure if you block new tables the test will fail
2. `columns` are lambdas and new columns are created from data
yes - there's a little overlap in (2) with what normalizer does and it will not happen often but we need to handle this